File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/eal_to_am.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n, m;
    cout << "Nhap so dinh: ";
    cin >> n;
    cout << "Nhap so canh: ";
    cin >> m;

    // Khoi tao ma tran ke n x n toan 0
    vector<vector<int>> am(n, vector<int>(n, 0));

    cout << "Nhap danh sach ke mo rong (u v w):\n";
    for (int i = 0; i < m; ++i) {
        int u, v, w;
        cin >> u >> v >> w;
        am[u][v] = w;  // Gan so luong canh tu u den v
    }

    // In ma tran ke
    cout << "Ma tran ke (AM):\n";
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            cout << am[i][j] << " ";
        }
        cout << "\n";
    }

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/eal_to_am.py
# Nhap so dinh va so canh
n = int(input("Nhap so dinh: "))
m = int(input("Nhap so canh: "))

# Khoi tao ma tran ke n x n toan 0
am = [[0 for _ in range(n)] for _ in range(n)]

print("Nhap danh sach ke mo rong (u v w):")
for _ in range(m):
    u, v, w = map(int, input().split())
    am[u][v] = w  # Gan so luong canh tu u den v

# In ma tran ke
print("Ma tran ke (AM):")
for i in range(n):
    print(' '.join(map(str, am[i])))
